gpt_synthesizer/generate_main.py

```diff
@@ -0,0 +1,106 @@
+from langchain.chains import LLMChain
+from langchain.prompts.prompt import PromptTemplate
+import os
+from prompt import get_generate_main_prompt
+from gpt_synthesizer.parser import get_code_from_chat
+import logging
+from ui import GEN_MAIN_MSG
+
+
+def to_files(chat, chat_key):
+    workspace = dict()
+
+    workspace[chat_key] = chat
+
+    files = get_code_from_chat(chat)
+    for file_name, code in files:
+        workspace[file_name] = code
+
+    return workspace
+
+
+def main_generator(task, language, component_list, summary, llm, verbose=False):
+    # Get the contents of the workspace
+    folder_path = '../workspace/'
+
+    # List all files in the folder
+    file_list = os.listdir(folder_path)
+
+    # Iterate through the files and read their contents
+    total_contents = ''
+
+    for filename in file_list:
+        file_path = os.path.join(folder_path, filename)
+        if os.path.isfile(file_path) and filename.endswith('.py'):  # Ensure it's a file (not a subdirectory, etc.)
+            print(f"Implementation of {filename}")
+            with open(file_path, 'r') as file:
+                file_contents = file.read()
+                #print(f"Contents of {filename}:\n{file_contents}\n")
+                total_contents += f"### {filename}\n{file_contents}\n"
+    # print(f"Total contents:\n{total_contents}")
+
+    gen_code_prompt = get_generate_main_prompt()
+
+    gen_code_chain = LLMChain(
+        llm=llm,
+        prompt=gen_code_prompt,
+        verbose=verbose
+    )
+
+    gen_code_output = gen_code_chain.predict(task=task, language=language, component_list=component_list,
+                                             summary=summary, total_contents=total_contents)
+
+    chat_key = "all_output.txt"
```

Comment:
Since this part is repeated in generate_code.py, maybe it would be a better idea to move this to to_files().